numba_dpex/kernel_api/atomic_ref.py

```diff
@@ -200,3 +200,55 @@ def exchange(self, val):
         old = self._ref[self._index]
         self._ref[self._index] = val
         return old
+
+    def compare_exchange_weak(self, expected, desired, expected_idx=0):
```

Comment:
I get the rationale, and having discussed it with you I is clear to me that `expected` needs to be an Array type. However, I still feel that from a general usage purpose it is rather confusing. We should create a `Ref` type in numba-dpex to make the usage clear.

Reply:
Created #1316 for tracking work on new array ref type, which would be handled in a separate PR.